openwebwork/webwork2
# Restore sendXMLRPC.pl changes

Fixes a problem with -v (verbose) and adds two features.

    You can specify the problemSeed using the option -seed=s where s is a number
    You can take input from STDIN using sendXMLRPC.pl -h - (- stands for STDIN)

To test: in the directory webwork2/clients
1.

sendXMLRPC.pl -v (before patch this will do nothing)
after patch this will print out information telling you which credential file is being used.

sendXMLRPC.pl --seed=123 -b t/input.pg
inspect the line at the top of the problem to make sure that the problemSeed printed there agrees with the seed you entered.

    cat t/input.pg |sendXMLRPC --seed=432 -b -
    should work the same as the command above (except the problemSeed is different)

    If you have this set up in a docker box and have vim installed in the docker box you can try
    vim t/input.pg
    inside vim type
    :w ! ./pr
    This should send an html version of the vim buffer to /var/www/html/xmlrpc_out.html
    Using the browser view the rendered version at http://localhost:8080/xmlrpc_out.html

By editing the script pr you could also use this to print the vim buffer outside the docker environment -- just direct the output of sendXMLRPC.pl to the appropriate location or application
